[CI] XPackRestIT test {p0=transform/transforms_start_stop/Test stop all} failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/pky3bwfrwrpjs/tests/:x-pack:plugin:yamlRestTest/org.elasticsearch.xpack.test.rest.XPackRestIT/test%20%7Bp0=transform%2Ftransforms_start_stop%2FTest%20stop%20all%7D

**Reproduction line:**
`./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=transform/transforms_start_stop/Test stop all}" -Dtests.seed=1F8B9F800B9DEB00 -Dtests.locale=he-IL -Dtests.timezone=US/Michigan -Druntime.java=17`

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.test.rest.XPackRestIT&tests.test=test%20%7Bp0%3Dtransform/transforms_start_stop/Test%20stop%20all%7D

**Failure excerpt:**
```
java.lang.AssertionError: Failure at [transform/transforms_start_stop:373]: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:108)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:104)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$2(TransportStopTransformAction.java:241)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:108)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:104)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$2(TransportStopTransformAction.java:241)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]

  at __randomizedtesting.SeedInfo.seed([1F8B9F800B9DEB00:97DFA05AA56186F8]:0)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:510)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:479)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.lang.AssertionError: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:108)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:104)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$2(TransportStopTransformAction.java:241)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:108)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:104)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$2(TransportStopTransformAction.java:241)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]

    at org.junit.Assert.fail(Assert.java:88)
    at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:373)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:499)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:479)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:568)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:833)

```

I think this failure https://gradle-enterprise.elastic.co/s/stnyrb2pqpidq is related:
```
REPRODUCE WITH: ./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=transform/transforms_start_stop/Test start/stop/start continuous transform}" -Dtests.seed=BB0F32B044712F91 -Dtests.locale=sk-SK -Dtests.timezone=Asia/Yakutsk -Druntime.java=17

org.elasticsearch.xpack.test.rest.XPackRestIT > test {p0=transform/transforms_start_stop/Test start/stop/start continuous transform} FAILED
    java.lang.AssertionError: Failure at [transform/transforms_start_stop:233]: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]
```

The stacktrace points into failure handling code to build a proper error message. For me it looks like that there is a bug within that bug. It contains `buildException(TransportStopTransformAction.java:323)`, which is in the middle of that method. I recently added asserts, but they haven't triggered either.

I am going to add some debug logging.

Full trace:

```
08:24:47   1> [2022-08-03T15:20:30,720][INFO ][o.e.x.t.r.XPackRestIT    ] [test] [p0=transform/transforms_start_stop/Test start/stop/start continuous transform] after test
08:24:47   2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=transform/transforms_start_stop/Test start/stop/start continuous transform}" -Dtests.seed=BB0F32B044712F91 -Dtests.locale=sk-SK -Dtests.timezone=Asia/Yakutsk -Druntime.java=17
08:24:47   2> java.lang.AssertionError: Failure at [transform/transforms_start_stop:233]: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]
08:24:47         at __randomizedtesting.SeedInfo.seed([BB0F32B044712F91:335B0D6AEA8D4269]:0)
08:24:47         at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:510)
08:24:47         at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:479)
08:24:47         at jdk.internal.reflect.GeneratedMethodAccessor14.invoke(Unknown Source)
08:24:47         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:24:47         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:24:47         at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
08:24:47         at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
08:24:47         at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
08:24:47         at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
08:24:47         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
08:24:47         at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
08:24:47         at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
08:24:47         at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
08:24:47         at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
08:24:47         at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
08:24:47         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
08:24:47         at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
08:24:47         at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
08:24:47         at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
08:24:47         at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
08:24:47         at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
08:24:47         at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
08:24:47         at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
08:24:47         at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
08:24:47         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
08:24:47         at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
08:24:47         at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
08:24:47         at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
08:24:47         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
08:24:47         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
08:24:47         at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
08:24:47         at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
08:24:47         at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
08:24:47         at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
08:24:47         at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
08:24:47         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
08:24:47         at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
08:24:47         at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
08:24:47         at java.base/java.lang.Thread.run(Thread.java:833)
08:24:47 
08:24:47         Caused by:
08:24:47         java.lang.AssertionError: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:323)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1380)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1465)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1445)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:808)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]
08:24:47             at org.junit.Assert.fail(Assert.java:88)
08:24:47             at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:373)
08:24:47             at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:499)
08:24:47             ... 38 more
```

I found a PR build that contains the debug log and a potentially useful st:

```
[2022-08-23T09:12:28,429][DEBUG][o.e.x.t.a.TransportStopTransformAction] [yamlRestTest-0] [airline-transform-start-stop] Failure when waiting for transform to stop, task failures: [[[EPPCTzNISI-slKEeCqJKQg][58387] failed, reason [org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop] state value should_stop_at_checkpoint from [false] to [false]]]], node failures: [[]]
[2022-08-23T09:12:28,428][ERROR][o.e.x.t.t.TransformTask  ] [yamlRestTest-0] [airline-transform-start-stop] failed updating state to [{"task_state":"started","indexer_state":"stopped","checkpoint":0,"progress":{"docs_indexed":0,"docs_processed":0},"should_stop_at_checkpoint":false}].
java.util.NoSuchElementException: null
	at java.util.TreeMap.key(TreeMap.java:1602) ~[?:?]
	at java.util.TreeMap.firstKey(TreeMap.java:291) ~[?:?]
	at java.util.TreeSet.first(TreeSet.java:391) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.scheduling.TransformScheduledTaskQueue.first(TransformScheduledTaskQueue.java:57) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.scheduling.TransformScheduler.processScheduledTasksInternal(TransformScheduler.java:137) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.scheduling.TransformScheduler.processScheduledTasks(TransformScheduler.java:110) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.scheduling.TransformScheduler.registerTransform(TransformScheduler.java:203) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.TransformTask.lambda$start$1(TransformTask.java:278) ~[?:?]
	at org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.xpack.transform.transforms.TransformTask.lambda$persistStateToClusterState$5(TransformTask.java:460) ~[?:?]
	at org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener$MappedActionListener.onResponse(ActionListener.java:127) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:31) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.client.internal.node.NodeClient$SafelyWrappedActionListener.onResponse(NodeClient.java:160) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:207) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:201) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:31) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.lambda$applyInternal$2(SecurityActionFilter.java:165) ~[?:?]
	at org.elasticsearch.action.ActionListener$DelegatingFailureActionListener.onResponse(ActionListener.java:245) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener$DelegatingActionListener.onResponse(ActionListener.java:212) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener$MappedActionListener.onResponse(ActionListener.java:127) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.persistent.PersistentTasksClusterService$4.clusterStateProcessed(PersistentTasksClusterService.java:274) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.lambda$execute$0(MasterService.java:553) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$ExecutionResult.onPublishSuccess(MasterService.java:959) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$2.onResponse(MasterService.java:388) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$2.onResponse(MasterService.java:383) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener.completeWith(ActionListener.java:473) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService.publish(MasterService.java:478) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService.publishClusterStateUpdate(MasterService.java:367) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:304) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:170) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:110) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:148) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:820) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:257) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:223) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[2022-08-23T09:12:28,431][ERROR][o.e.x.t.t.TransformPersistentTasksExecutor] [yamlRestTest-0] Failed to start task [airline-transform-start-stop] in node operation
org.elasticsearch.ElasticsearchException: Error while updating state for transform [airline-transform-start-stop] to [STOPPED].
	at org.elasticsearch.xpack.transform.transforms.TransformTask.lambda$start$3(TransformTask.java:289) ~[?:?]
	at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.xpack.transform.transforms.TransformTask.lambda$persistStateToClusterState$5(TransformTask.java:460) ~[?:?]
	at org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener$MappedActionListener.onResponse(ActionListener.java:127) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:31) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.client.internal.node.NodeClient$SafelyWrappedActionListener.onResponse(NodeClient.java:160) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:207) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:201) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:31) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.lambda$applyInternal$2(SecurityActionFilter.java:165) ~[?:?]
	at org.elasticsearch.action.ActionListener$DelegatingFailureActionListener.onResponse(ActionListener.java:245) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener$DelegatingActionListener.onResponse(ActionListener.java:212) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener$MappedActionListener.onResponse(ActionListener.java:127) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.persistent.PersistentTasksClusterService$4.clusterStateProcessed(PersistentTasksClusterService.java:274) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.lambda$execute$0(MasterService.java:553) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$ExecutionResult.onPublishSuccess(MasterService.java:959) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$2.onResponse(MasterService.java:388) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$2.onResponse(MasterService.java:383) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.action.ActionListener.completeWith(ActionListener.java:473) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService.publish(MasterService.java:478) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService.publishClusterStateUpdate(MasterService.java:367) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:304) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:170) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:110) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:148) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:820) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:257) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:223) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.util.NoSuchElementException
	at java.util.TreeMap.key(TreeMap.java:1602) ~[?:?]
	at java.util.TreeMap.firstKey(TreeMap.java:291) ~[?:?]
	at java.util.TreeSet.first(TreeSet.java:391) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.scheduling.TransformScheduledTaskQueue.first(TransformScheduledTaskQueue.java:57) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.scheduling.TransformScheduler.processScheduledTasksInternal(TransformScheduler.java:137) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.scheduling.TransformScheduler.processScheduledTasks(TransformScheduler.java:110) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.scheduling.TransformScheduler.registerTransform(TransformScheduler.java:203) ~[?:?]
	at org.elasticsearch.xpack.transform.transforms.TransformTask.lambda$start$1(TransformTask.java:278) ~[?:?]
	at org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162) ~[elasticsearch-8.5.0-SNAPSHOT.jar:?]
	... 30 more
	
```

--- a/docs/changelog/89716.yaml
+++ b/docs/changelog/89716.yaml
@@ -1,0 +1,6 @@
+pr: 89716
+summary: Scheduler concurrency fix
+area: Transform
+type: bug
+issues:
+ - 88991

--- a/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduledTaskQueue.java
+++ b/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduledTaskQueue.java
@@ -44,16 +44,14 @@ class TransformScheduledTaskQueue {
     }
 
     /**
-     * @return whether the queue is empty.
-     */
-    public synchronized boolean isEmpty() {
-        return tasks.isEmpty();
-    }
-
-    /**
-     * @return the task with the *lowest* priority.
+     * @return the task with the *lowest* priority or null if the queue is empty.
      */
     public synchronized TransformScheduledTask first() {
+        // gh#88991 concurrent access: the empty check must run within the synchronized context
+        if (tasks.isEmpty()) {
+            return null;
+        }
+
         return tasks.first();
     }
 

--- a/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduler.java
+++ b/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduler.java
@@ -129,12 +129,14 @@ public final class TransformScheduler {
     }
 
     private boolean processScheduledTasksInternal() {
-        if (scheduledTasks.isEmpty()) {
+        TransformScheduledTask scheduledTask = scheduledTasks.first();
+
+        if (scheduledTask == null) {
             // There are no scheduled tasks, hence, nothing to do
             return false;
         }
         long currentTimeMillis = clock.millis();
-        TransformScheduledTask scheduledTask = scheduledTasks.first();
+
         // Check if the task is eligible for processing
         if (currentTimeMillis < scheduledTask.getNextScheduledTimeMillis()) {
             // It is too early to process this task.

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduledTaskQueueTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduledTaskQueueTests.java
@@ -17,7 +17,6 @@ import org.junit.Before;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
-import java.util.NoSuchElementException;
 import java.util.Set;
 import java.util.concurrent.Callable;
 import java.util.concurrent.Future;
@@ -27,7 +26,8 @@ import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.is;
-import static org.junit.Assert.fail;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
 
 public class TransformScheduledTaskQueueTests extends ESTestCase {
 
@@ -52,14 +52,14 @@ public class TransformScheduledTaskQueueTests extends ESTestCase {
 
     public void testNonEmptyQueue() {
         queue.add(createTask("task-1", 5));
-        assertThat(queue.isEmpty(), is(false));
+        assertThat(queue.first(), is(notNullValue()));
     }
 
     public void testAddAndRemove() {
         queue.add(createTask("task-1", 5));
         queue.add(createTask("task-2", 1));
         queue.add(createTask("task-3", 9));
-        assertThat(queue.isEmpty(), is(false));
+        assertThat(queue.first(), is(notNullValue()));
         assertThat(queue.getTransformIds(), containsInAnyOrder("task-1", "task-2", "task-3"));
         assertThat(queue.first(), is(equalTo(createTask("task-2", 1))));
 
@@ -83,7 +83,7 @@ public class TransformScheduledTaskQueueTests extends ESTestCase {
                 assertThat(taskAdded, is(true));
             }
         }
-        assertThat(queue.isEmpty(), is(false));
+        assertThat(queue.first(), is(notNullValue()));
         assertThat(queue.getTransformIds(), hasSize(100));
 
         {
@@ -117,7 +117,7 @@ public class TransformScheduledTaskQueueTests extends ESTestCase {
         queue.add(createTask("task-1", 5));
         queue.remove("task-non-existent");
         // Verify that the remove operation had no effect
-        assertThat(queue.isEmpty(), is(false));
+        assertThat(queue.first(), is(notNullValue()));
         assertThat(queue.getTransformIds(), containsInAnyOrder("task-1"));
         assertThat(queue.first(), is(equalTo(createTask("task-1", 5))));
     }
@@ -126,7 +126,7 @@ public class TransformScheduledTaskQueueTests extends ESTestCase {
         queue.add(createTask("task-1", 5));
         queue.update("task-non-existent", task -> createTask(task.getTransformId(), -999));
         // Verify that the update operation had no effect
-        assertThat(queue.isEmpty(), is(false));
+        assertThat(queue.first(), is(notNullValue()));
         assertThat(queue.getTransformIds(), containsInAnyOrder("task-1"));
         assertThat(queue.first(), is(equalTo(createTask("task-1", 5))));
     }
@@ -147,7 +147,7 @@ public class TransformScheduledTaskQueueTests extends ESTestCase {
         queue.add(createTask("task-7", 0));
         queue.add(createTask("task-8", 2));
         queue.add(createTask("task-9", 4));
-        assertThat(queue.isEmpty(), is(false));
+        assertThat(queue.first(), is(notNullValue()));
         assertThat(
             queue.getTransformIds(),
             containsInAnyOrder("task-1", "task-2", "task-3", "task-4", "task-5", "task-6", "task-7", "task-8", "task-9")
@@ -155,7 +155,7 @@ public class TransformScheduledTaskQueueTests extends ESTestCase {
         assertThat(queue.first(), is(equalTo(createTask("task-7", 0))));
 
         List<TransformScheduledTask> tasksByPriority = new ArrayList<>();
-        while (queue.isEmpty() == false) {
+        while (queue.first() != null) {
             TransformScheduledTask task = queue.first();
             tasksByPriority.add(task);
             queue.remove(task.getTransformId());
@@ -210,8 +210,7 @@ public class TransformScheduledTaskQueueTests extends ESTestCase {
     }
 
     private void assertThatQueueIsEmpty() {
-        assertThat(queue.isEmpty(), is(true));
+        assertThat(queue.first(), is(nullValue()));
         assertThat(queue.getTransformIds(), is(empty()));
-        expectThrows(NoSuchElementException.class, () -> queue.first());
     }
 }

